Voycar.Api.Web/Features/Members/Post/Login/Endpoint.cs

```diff
@@ -29,51 +29,47 @@ public override void Configure()
 
     public override async Task HandleAsync(Request req, CancellationToken ct)
     {
-        Member? member = null;
-
-        var user = await this._userRepository.Retrieve("email", req.Email.ToLowerInvariant());
+        var user = await this._userRepository.Retrieve("email", req.Email!.ToLowerInvariant());
 
-        // Check if user is a member (members must be verified)
-        if (user is not null)
-        {
-            member = this._members.Retrieve(user.Id);
-        }
 
-        // Checks for employee / admin
-        if (member is null)
-        {
-            // Check if employee or admin entered valid credentials
-            if (user is null || !BCrypt.Net.BCrypt.EnhancedVerify(req.Password, user.PasswordHash))
-            {
-                await this.SendErrorsAsync(cancellation: ct);
-                return;
-            }
-
-            // Login employee / admin
-            await this.SignInUserAsync(user!, ct);
-            return;
-        }
-
-        // Check if member entered valid credentials and is verified
-        if (member!.VerifiedAt is null || !BCrypt.Net.BCrypt.EnhancedVerify(req.Password, member.User.PasswordHash))
+        // Check if user entered valid credentials and verified his email
+        if (user is null || user!.VerifiedAt is null ||
+            !BCrypt.Net.BCrypt.EnhancedVerify(req.Password, user.PasswordHash))
         {
+            // Do not send different request for user not found, since this would reveal which users are signed up
             await this.SendErrorsAsync(cancellation: ct);
             return;
         }
 
         // Login member
-        await this.SignInUserAsync(user!, ct);
+        await this.SignInUserAsync(user!);
     }
 
 
-    private async Task SignInUserAsync(User user, CancellationToken ct)
+    /// <summary>
+    /// Signs in a user with cookie authentication and assigns any possible roles.
+    /// If any roles are associated with that user the roles are added to the cookie. A response including the
+    /// cookie is sent back.
+    /// </summary>
+    /// <param name="user">The user to sign in</param>
+    private async Task SignInUserAsync(User user)
     {
-        var role = await this._members.RetrieveRole(user.RoleId);
-        await CookieAuth.SignInAsync(u =>
+        var roleId = user.RoleId;
+        if (roleId is null)
+        {
+            // Sign in without any roles
+            await CookieAuth.SignInAsync(privileges => { });
```

Comment:
Wird es vorkommen, dass ein User keine RoleID hat? Wenn man sich registriert hat, dann kann die RoleID eigentlich nicht null sein? Oder ist hiermit noch was anderes geplant?

Reply:
Hmmm, ja das stimmt eigentlich. Das kam noch von der Aufteilung des Registrierungsendpunktes in User und Member Registrierung.
Dann mache ich Rolle ein nicht `nullable` Feld im User und passe den Login an.

Reply:
6292549